internal/handshake/params_negotiator.go

```diff
@@ -0,0 +1,104 @@
+package handshake
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+)
+
+type paramsNegotiator struct {
+	paramsNegotiatorBase
+}
+
+var _ ParamsNegotiator = &paramsNegotiator{}
+
+// newParamsNegotiator creates a new connection parameters manager
+func newParamsNegotiator(pers protocol.Perspective, v protocol.VersionNumber, params *TransportParameters) *paramsNegotiator {
+	h := &paramsNegotiator{}
+	h.perspective = pers
+	h.version = v
+	h.init(params)
+	return h
+}
+
+func (h *paramsNegotiator) SetFromTransportParameters(params []transportParameter) error {
```

Comment:
Do we have to lock the mutex here?

Reply:
Yeah, I missed that. Same for `GetTransportParameters`.